cas-server-core-authentication/src/main/java/org/apereo/cas/config/CasCoreAuthenticationConfiguration.java

```diff
@@ -0,0 +1,402 @@
+import org.apereo.cas.authentication.DefaultPrincipalElectionStrategy;
+import org.apereo.cas.authentication.FileTrustStoreSslSocketFactory;
+import org.apereo.cas.authentication.NotPreventedAuthenticationPolicy;
+import org.apereo.cas.authentication.PolicyBasedAuthenticationManager;
+import org.apereo.cas.authentication.PrincipalElectionStrategy;
+import org.apereo.cas.authentication.RegisteredServiceAuthenticationHandlerResolver;
+import org.apereo.cas.authentication.RequiredHandlerAuthenticationPolicy;
+import org.apereo.cas.authentication.RequiredHandlerAuthenticationPolicyFactory;
+import org.apereo.cas.authentication.SuccessfulHandlerMetaDataPopulator;
+import org.apereo.cas.authentication.handler.ConvertCasePrincipalNameTransformer;
+import org.apereo.cas.authentication.handler.DefaultPasswordEncoder;
+import org.apereo.cas.authentication.handler.NoOpPrincipalNameTransformer;
+import org.apereo.cas.authentication.handler.PasswordEncoder;
+import org.apereo.cas.authentication.handler.PlainTextPasswordEncoder;
+import org.apereo.cas.authentication.handler.PrefixSuffixPrincipalNameTransformer;
+import org.apereo.cas.authentication.handler.PrincipalNameTransformer;
+import org.apereo.cas.authentication.handler.support.HttpBasedServiceCredentialsAuthenticationHandler;
+import org.apereo.cas.authentication.handler.support.JaasAuthenticationHandler;
+import org.apereo.cas.authentication.principal.BasicPrincipalResolver;
+import org.apereo.cas.authentication.principal.DefaultPrincipalFactory;
+import org.apereo.cas.authentication.principal.PersonDirectoryPrincipalResolver;
+import org.apereo.cas.authentication.principal.PrincipalFactory;
+import org.apereo.cas.authentication.principal.PrincipalResolver;
+import org.apereo.cas.authentication.principal.RememberMeAuthenticationMetaDataPopulator;
+import org.apereo.cas.authentication.support.PasswordPolicyConfiguration;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.apereo.cas.services.ReloadableServicesManager;
+import org.apereo.cas.util.http.HttpClient;
+import org.apereo.cas.web.flow.AuthenticationExceptionHandler;
+import org.apereo.services.persondir.IPersonAttributeDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.Resource;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * This is {@link CasCoreAuthenticationConfiguration}.
+ *
+ * @author Misagh Moayyed
+ * @author Dmitriy Kopylenko
+ * @since 5.0.0
+ */
+@Configuration("casCoreAuthenticationConfiguration")
+public class CasCoreAuthenticationConfiguration {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [This class has too many methods, consider refactoring it.](https://www.codacy.com/app/mmoayyed/cas/file/2768422476/issues/source?bid=3256115&fileBranchId=3357337#l72)